ssl/statem/statem_lib.c

```diff
@@ -2128,6 +2128,16 @@ int check_in_list(SSL *s, uint16_t group_id, const uint16_t *groups,
     if (groups == NULL || num_groups == 0)
         return 0;
 
+    if (checkallow) {
+        if (group_id >= TLSEXT_GROUPID_brainpoolP256r1
+                && group_id <= TLSEXT_GROUPID_brainpoolP512r1)
+            return 0;
+        if (group_id >= TLSEXT_GROUPID_brainpoolP256r1_tls13
+                && group_id <= TLSEXT_GROUPID_brainpoolP512r1_tls13)
+            group_id = group_id - TLSEXT_GROUPID_brainpoolP256r1_tls13
+                                + TLSEXT_GROUPID_brainpoolP256r1;
+    }
```

Comment:
I really don't like this. I think you really should just add this like you would add a normal curve.

I also suggest that we add a minimum / maximum version for curves, I think that should simplify things.